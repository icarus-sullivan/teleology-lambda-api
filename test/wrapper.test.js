import { wrapper, ApiError } from '../src';

const event = {
  pathParameters: {
    slug: 'test',
  },
  queryStringParameters: {
    next: 10,
  },
};

describe.only('wrapper', () => {
  it('success defaults', async () => {
    const handler = () => ({
      foo: 'bar',
    });

    const result = await wrapper(handler)(event);
    expect(result).toMatchSnapshot();
  });

  it('normal response if nothing returned', async () => {
    const handler = () => undefined;

    const result = await wrapper(handler)(event);
    expect(result).toMatchSnapshot();
  });

  it('success custom headers and body', async () => {
    const handler = () => ({
      headers: {
        'Content-Type': 'text/html',
      },
      body: '<h1>hello</h1>',
    });

    const result = await wrapper(handler)(event);
    expect(result).toMatchSnapshot();
  });

  it('error defaults', async () => {
    const handler = () => {
      throw new Error('foo');
    };

    const result = await wrapper(handler)(event);
    expect(result).toMatchSnapshot();
  });

  it('custom error', async () => {
    const handler = () => {
      throw new ApiError('Unauthorized', { code: 401, description: 'nope' });
    };

    const result = await wrapper(handler)(event);
    expect(result).toMatchSnapshot();
  });
});
