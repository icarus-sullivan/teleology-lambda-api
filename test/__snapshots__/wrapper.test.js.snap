// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wrapper custom error 1`] = `
Object {
  "body": "{\\"name\\":\\"ApiError\\",\\"message\\":\\"Unauthorized\\",\\"code\\":401,\\"description\\":\\"nope\\"}",
  "headers": Object {
    "Access-Control-Allow-Credentials": true,
    "Access-Control-Allow-Origin": "*",
    "Content-Type": "application/json",
  },
  "statusCode": "401",
}
`;

exports[`wrapper error defaults 1`] = `
Object {
  "body": "{\\"message\\":\\"foo\\"}",
  "headers": Object {
    "Access-Control-Allow-Credentials": true,
    "Access-Control-Allow-Origin": "*",
    "Content-Type": "application/json",
  },
  "statusCode": "500",
}
`;

exports[`wrapper normal response if nothing returned 1`] = `
Object {
  "body": "{}",
  "headers": Object {
    "Access-Control-Allow-Credentials": true,
    "Access-Control-Allow-Origin": "*",
    "Content-Type": "application/json",
  },
  "statusCode": "200",
}
`;

exports[`wrapper success custom headers and body 1`] = `
Object {
  "body": "<h1>hello</h1>",
  "headers": Object {
    "Access-Control-Allow-Credentials": true,
    "Access-Control-Allow-Origin": "*",
    "Content-Type": "text/html",
  },
  "statusCode": "200",
}
`;

exports[`wrapper success defaults 1`] = `
Object {
  "body": "{\\"foo\\":\\"bar\\"}",
  "headers": Object {
    "Access-Control-Allow-Credentials": true,
    "Access-Control-Allow-Origin": "*",
    "Content-Type": "application/json",
  },
  "statusCode": "200",
}
`;
